### DISCUSSION TITLE
lxc容器，使用host模式，安装多个青龙，只有第一个能访问

### DISCUSSION THREAD

**Discussion by ZJBOY1314** [whyour/qinglong#2271] [Q&A]
lxc容器，使用host模式，安装多个青龙，只有第一个能访问，不知道为啥。
我按照要求修改了Port还是一样的，不知道什么原因
docker run -dit \
  -v $PWD/ql/data:/ql/data \
  -p 5701:5701 \
  -e QlBaseUrl="/root/ql1" \
  -e QlPort="5701" \
  --name ql1 \
  --hostname ql1 \
  --net=host \
  --restart unless-stopped \
  whyour/qinglong:latest

**Comment by luoyexiaoyao** [comment]
挂载的目录换一下试试?

**Comment by ZJBOY1314** [comment]
请问能提供个案例不哦，大佬，哭了，这是我现在使用的脚本，不知道怎么改。😭😭😭😭😭😭😭😭😭

docker&nbsp;run&nbsp;-dit&nbsp;\
&nbsp;&nbsp;&nbsp;-v&nbsp;$PWD/ql1/config:/ql/config&nbsp;\
&nbsp;&nbsp;&nbsp;-v&nbsp;$PWD/ql1/log:/ql/log&nbsp;\
&nbsp;&nbsp;&nbsp;-v&nbsp;$PWD/ql1/db:/ql/db&nbsp;\
&nbsp;&nbsp;&nbsp;-v&nbsp;$PWD/ql1/repo:/ql/repo&nbsp;\
&nbsp;&nbsp;&nbsp;-v&nbsp;$PWD/ql1/raw:/ql/raw&nbsp;\
&nbsp;&nbsp;&nbsp;-v&nbsp;$PWD/ql1/scripts:/ql/scripts&nbsp;\
&nbsp;&nbsp;&nbsp;-v&nbsp;$PWD/ql1/jbot:/ql/jbot&nbsp;\
&nbsp;&nbsp;&nbsp;-p&nbsp;5701:5701&nbsp;\
&nbsp;&nbsp;&nbsp;--name&nbsp;ql1&nbsp;\
&nbsp;&nbsp;&nbsp;--hostname&nbsp;ql1&nbsp;\
&nbsp;&nbsp;&nbsp;--restart&nbsp;unless-stopped&nbsp;\
&nbsp;&nbsp;&nbsp;--net=host&nbsp;&nbsp;\
&nbsp;&nbsp;&nbsp;-e&nbsp;QlPort=&quot;5701&quot;&nbsp;\
&nbsp;&nbsp;&nbsp;whyour/qinglong:2.15.16


---原始邮件---
发件人: ***@***.***&gt;
发送时间: 2024年2月22日(周四) 中午1:59
收件人: ***@***.***&gt;;
抄送: ***@***.******@***.***&gt;;
主题: Re: [whyour/qinglong] lxc容器，使用host模式，安装多个青龙，只有第一个能访问 (Discussion #2271)




 
挂载的目录换一下试试?
 
&mdash;

**Reply by luoyexiaoyao** [reply]
唔 咱之后了解到的青龙面板好像不止使用了一个端口,可能是5400 5500 5600(后端)5700(前端)
可以尝试一下把/ql/.env /ql/shell/所有文件
/ql/docker/所有文件的默认端口全部修改一下
下边的代码在容器中运行可能可以快速修改端口为默认+1
```
sed -i 's/5400/5401/; s/5500/5501/; s/5600/5601/; s/5700/5701/' /ql/.env
sed -i 's/5400/5401/; s/5500/5501/; s/5600/5601/; s/5700/5701/' /ql/shell/*
sed -i 's/5400/5401/; s/5500/5501/; s/5600/5601/; s/5700/5701/; s/5701:5700/5701:5701/' /ql/docker/*
```
之后再重启容器试试

**Comment by ZJBOY1314** [comment]
使用指令 
docker exec -it <container_name_or_id&gt; /bin/bash进入容器执行您说的


sed -i 's/5400/5401/; s/5500/5501/; s/5600/5601/; s/5700/5701/' /ql/.env 
sed -i 's/5400/5401/; s/5500/5501/; s/5600/5601/; s/5700/5701/' /ql/shell/* 
sed -i 's/5400/5401/; s/5500/5501/; s/5600/5601/; s/5700/5701/; s/5701:5700/5701:5701/' /ql/docker/* 
重启容器后，
尝试浏览器进入127.0.0.1:5701有反应了，提示 {启动中，请稍后。。。}   但是url进入的是127.0.0.1:5701/error，不知道为啥

---原始邮件---
发件人: ***@***.***&gt;
发送时间: 2024年2月26日(周一) 下午4:42
收件人: ***@***.***&gt;;
抄送: ***@***.******@***.***&gt;;
主题: Re: [whyour/qinglong] lxc容器，使用host模式，安装多个青龙，只有第一个能访问 (Discussion #2271)




 
唔 咱之后了解到的青龙面板好像不止使用了一个端口,可能是5400 5500 5600(后端)5700(前端)
 可以尝试一下把/ql/.env /ql/shell/所有文件
 /ql/docker/所有文件的默认端口全部修改一下
 下边的代码在容器中运行可能可以快速修改端口为默认+1
 sed -i 's/5400/5401/; s/5500/5501/; s/5600/5601/; s/5700/5701/' /ql/.env sed -i 's/5400/5401/; s/5500/5501/; s/5600/5601/; s/5700/5701/' /ql/shell/* sed -i 's/5400/5401/; s/5500/5501/; s/5600/5601/; s/5700/5701/; s/5701:5700/5701:5701/' /ql/docker/*  
之后再重启容器试试

**Reply by luoyexiaoyao** [reply]
大概因为ql初始化不成功 ,需要在容器内执行ql check重新……

**Comment by whyour** [comment]
host 模型不需要指定 -p，指定 QlPort 就行，每次要不一样
```
-e QlPort="5701"
```

**Comment by wings110** [comment]
lxc运行 qinglong  用bridge 模式是不是无法访问qinglong网页
